tests/test_utils.py

```diff
@@ -317,3 +319,170 @@ def check_hash_is_valid(dfh):
+
+            # parameters dictionary from `pd.read_excel` and `pd.read_csv` dataframes should be the same
+            assert excel_param_dict == csv_param_dict
+
+        def initialise_population(self, population):
+            pass
+
+        def initialise_simulation(self, sim):
+            pass
+
+    dummy_module = DummyModule()
+    dummy_module.read_parameters(Path(resourcefilepath / 'ResourceFile_test_convert_to_csv'))
+
+
+def test_convert_excel_files_method():
+    """ Test converting Excel files to csv equivalent is done as expected
+
+        1) Excel file name should become the name of the folder containing the newly converted csv files
+        2) Excel file sheet names should become csv file names
+        3) if files are given, the function should only convert to excel only those given files in a folder
+        4) if no files are given, all Excel files in the parent folder and subsequent folders within the parent folder
+           should get converted to csv files
+
+    """
+
+    def check_logic_of_converting_excel_files_to_csv_files(folder: Path, files: Optional[list[str]] = None) -> None:
+        """ check converting Excel files to csv files is done as expected
+                1) check that a new directory to hold the newly created csv files has been created
+                2) check that this new directory name matches the Excel file name it has been created from
+                3) check csv files are created and that the csv names should match sheet names of an Excel file they
+                have been created from
+        """
+        # check that the above function has created a folder named `ResourceFile_load-parameters`(name of the Excel
+        # file) and a csv file named `parameter_values` (Excel file sheet name).
+        excel_file_paths = [folder / file for file in files] \
+            if files is not None else [file for file in folder.rglob("*.xlsx")]
+
+        for excel_file_path in excel_file_paths:
+            xl = pd.ExcelFile(excel_file_path)
+            path_to_new_directory = Path(str(excel_file_path).split('.')[0])
+            # new folder should be created
+            assert path_to_new_directory.exists() and path_to_new_directory.is_dir()
+            # the new folder name should be the same as the Excel file name
+            assert excel_file_path.name.split('.')[0] == path_to_new_directory.name
+            for sheet_name in xl.sheet_names:
+                path_to_new_file = Path(path_to_new_directory / f'{sheet_name}.csv')
+                # new csv file(s) should be created with name(s) resembling sheet name(s) in excel file
+                assert path_to_new_file.exists() and path_to_new_file.is_file()
+                assert sheet_name == path_to_new_file.name.split('.')[0]
+
+            # remove the newly created folders and files to keep the resources folder clean
+            shutil.rmtree(Path(str(excel_file_path).split('.')[0]))
```

Comment:
I'd be very wary of using `shutil.rmtree` in tests, particularly with a programmatically constructed path, as it's difficult to predict where tests will be run from and whether this will actually delete the files you want to or something else unintended. For example if this line was run for a value of `excel_file_path` which contains a directory along the path with a name prefixed with `.` (a common pattern for hidden directories in Unix systems), then this line will attempt to remove the parent of that directory and all of its sub files / directories!

A safer pattern here would be to use [the `tmp_path` fixture ](https://docs.pytest.org/en/stable/how-to/tmp_path.html) and copy the relevant files required for conversion there and then rely the automatic clean up of the files in the `tmp_path` fixture directory at the end of the test function.

Reply:
Thanks. I have changed to now using temporal directory.